projects/js-packages/components/components/icon-tooltip/index.tsx

```diff
@@ -0,0 +1,97 @@
+import { Popover } from '@wordpress/components';
+import { useDebounce } from '@wordpress/compose';
+import classNames from 'classnames';
+import React, { useState } from 'react';
+import Gridicon from '../gridicon/index';
+import { IconTooltipProps, Placement, Position } from './types';
+
+import './style.scss';
+
+const placementsToPositions = ( placement: Placement ): Position => {
+	const mapping = {
+		'top-end': 'top left',
+		top: 'top center',
+		'top-start': 'top right',
+		'bottom-end': 'bottom left',
+		bottom: 'bottom center',
+		'bottom-start': 'bottom right',
+	};
+
+	return mapping[ placement ] as Position;
+};
+
+/**
+ * Generate Icon Tooltip
+ *
+ * @param {IconTooltipProps} props - Props
+ * @returns {React.ReactElement} - JSX element
+ */
+const IconTooltip: React.FC< IconTooltipProps > = ( {
+	className = '',
+	iconClassName = '',
+	placement = 'bottom-end',
+	animate = true,
+	iconCode = 'info-outline',
+	title,
+	children,
+} ) => {
+	const delay = 300;
+	const [ isVisible, setIsVisible ] = useState( false );
+	const delayedSetIsOver = useDebounce( setIsVisible, delay );
+
+	const createToggleIsOver = ( eventName, isDelayed = false ) => {
+		return event => {
+			event.stopPropagation();
+			event.preventDefault();
+			const _isVisible = [ 'focus', 'mouseenter' ].includes( event.type );
+			if ( _isVisible === isVisible ) {
+				return;
+			}
+
+			if ( isDelayed ) {
+				delayedSetIsOver( _isVisible );
+			} else {
+				setIsVisible( _isVisible );
+			}
+		};
+	};
+
+	const args = {
+		iconCode,
+		// To be compatible with deprecating option `position`
+		position: placementsToPositions( placement ),
```

Comment:
Can you tell me more about this? I suppose if an attribute is deprecated, we could make it transparent?

Reply:
Since [`placement` is announced](https://github.com/WordPress/gutenberg/tree/trunk/packages/components/src/popover#placement-top--top-start--top-end--right--right-start--right-end--bottom--bottom-start--bottom-end--left--left-start--left-end) as the replacement of legacy `position` in `Popover` component, I tend to use the newer prop and implicitly make it compatible with older version dependencies in this `IconTooltip` component. Even though the [newer version `Popover` component has processed the legacy prop](https://github.com/WordPress/gutenberg/blob/trunk/packages/components/src/popover/index.tsx#L244-L246) `position`.

In other words, users can only take care of `placement` no matter [what version of components](https://www.npmjs.com/package/@wordpress/components) they are using.